platform-metrics/src/main/java/org/hypertrace/core/serviceframework/metrics/PlatformMetricsRegistry.java

```diff
@@ -39,68 +64,216 @@
     final String fullMetricName = String.format("%s.%s", metricsPrefix, metricName);
     if (!METRIC_REGISTRY.getNames().contains(fullMetricName)) {
       METRIC_REGISTRY.register(fullMetricName, metric);
     }
   }
 
+  /**
+   * Registers a Counter with the given name with the service's metric registry and reports it
+   * periodically to the configured reporters. Apart from the given tags, the reporting service's
+   * default tags also will be reported with the metrics.
+   *
+   * See https://micrometer.io/docs/concepts#_counters for more details on the Counter.
+   */
+  public static Counter registerCounter(String name, Map<String, String> tags) {
+    return METER_REGISTRY.counter(name, addDefaultTags(tags));
+  }
+
+  /**
+   * Registers a Timer (without histograms) with the given name with the service's metric registry
+   * and reports it periodically to the configured reporters. Apart from the given tags, the
+   * reporting service's default tags also will be reported with the metrics.
+   *
+   * See https://micrometer.io/docs/concepts#_timers for more details on the Timer.
+   */
+  public static Timer registerTimer(String name, Map<String, String> tags) {
+    return registerTimer(name, tags, false);
+  }
+
+  /**
+   * Registers a Timer with the given name with the service's metric registry and reports it
+   * periodically to the configured reporters. Apart from the given tags, the reporting service's
+   * default tags also will be reported with the metrics.
+   *
+   * See https://micrometer.io/docs/concepts#_timers for more details on the Timer.
+   */
+  public static Timer registerTimer(String name, Map<String, String> tags, boolean histogram) {
+    Timer.Builder builder = Timer.builder(name)
+        .publishPercentiles(0.5, 0.95, 0.99)
+        .tags(addDefaultTags(tags));
+    if (histogram) {
+      builder = builder.publishPercentileHistogram();
+    }
+    return builder.register(METER_REGISTRY);
+  }
+
+  /**
+   * Registers the given number as a Guage with the service's metric registry and reports it
+   * periodically to the configured reporters. Apart from the given tags, the reporting service's
+   * default tags also will be reported with the metrics.
+   *
+   * See https://micrometer.io/docs/concepts#_gauges for more details on the Gauges.
+   */
+  public static <T extends Number> T registerGauge(String name, Map<String, String> tags, T number) {
+    return METER_REGISTRY.gauge(name, addDefaultTags(tags), number);
+  }
+
+  /**
+   * Registers metrics for the given Guava Cache with the service's metric registry and
+   * reports them periodically to the configured reporters. Apart from the given tags, the
+   * reporting service's default tags also will be reported with the metrics.
+   *
+   * See https://micrometer.io/docs/ref/cache for more details on the metrics.
+   */
+  public static void monitorGuavaCache(String name, Cache cache,
```

Comment:
Is this the reason for the guava dependency? Mixed feelings here, because that's a pretty heavy dependency to add for a use case that doesn't apply to every consumer here. On the other hand, guava is already an implicit dependency of grpc, so it may not really matter. Could move it to a separate module as an opt in - although not worth it if it doesn't remove the guava dependency from the service framework (don't remember if we already have a grpc dependency here).

Reply:
this module doesn't have grpc dependency, so Guava isn't already present here. I was on the fence about this change already so removed this now.
Anyways, the services can still get the registry and register caches with it. It's just that they can't use the default tags very easily, which also can be solved if needed.

Reply:
OK - we can address if needed